test/async-hooks/test-signalwrap.js

```diff
@@ -55,8 +55,12 @@ function onsigusr2() {
     process.on('SIGUSR2', common.mustCall(onsigusr2Again));
 
     const as = hooks.activitiesOfTypes('SIGNALWRAP');
-    assert.strictEqual(as.length, 2);
-    signal2 = as[1];
+    // TEST_THREAD_ID is set by tools/test.py. Adjust test results depending
+    // on whether the test was invoked via test.py or from the shell
+    // directly.
+    const expectedLength = process.env.TEST_THREAD_ID ? 2 : 3;
```

Comment:
Can I suggest updating the comment to explain that the third signal event listener is the SIGWINCH handler that node installs when it thinks `process.stdout` is a tty?

It's still kind of fragile because there might be a second SIGWINCH handler for `process.stderr`, i.e., a total of 4.

Reply:
@bnoordhuis Would the test be more robust by checking `process.stdout.isTTY` and `process.stderr.isTTY` rather than `process.env.TEST_THREAD_ID`?

Reply:
@trott I think that’s better, yes. There is only one async hook per active signal (which should probably be considered a bug), so currently the number is `2 + (process.stdout.isTTY || process.stderr.isTTY)`, I think.